editor/editor_file_system.cpp

```diff
@@ -293,24 +301,29 @@ void EditorFileSystem::_first_scan_process_scripts(const ScannedDirectory *p_sca
 				break;
 			}
 		}
-		if (!is_script) {
-			continue; // Not a script.
-		}
+		if (is_script) {
+			String path = p_scan_dir->full_path.path_join(scan_file);
+			String type = ResourceLoader::get_resource_type(path);
 
-		String path = p_scan_dir->full_path.path_join(scan_file);
-		String type = ResourceLoader::get_resource_type(path);
+			if (ClassDB::is_parent_class(type, SNAME("Script"))) {
+				String script_class_extends;
+				String script_class_icon_path;
+				String script_class_name = _get_global_script_class(type, path, &script_class_extends, &script_class_icon_path);
+				_register_global_class_script(path, path, type, script_class_name, script_class_extends, script_class_icon_path);
 
-		if (ClassDB::is_parent_class(type, SNAME("Script"))) {
-			String script_class_extends;
-			String script_class_icon_path;
-			String script_class_name = _get_global_script_class(type, path, &script_class_extends, &script_class_icon_path);
-			_register_global_class_script(path, path, type, script_class_name, script_class_extends, script_class_icon_path);
+				if (!script_class_name.is_empty()) {
+					p_existing_class_names.insert(script_class_name);
+				}
+			}
+		}
 
-			if (!script_class_name.is_empty()) {
-				p_existing_class_names.insert(script_class_name);
+		// Check for GDExtensions.
+		if (p_gdextension_extensions.find(ext)) {
+			String path = p_scan_dir->full_path.path_join(scan_file);
+			String type = ResourceLoader::get_resource_type(path);
```

Comment:
```suggestion
			const String path = p_scan_dir->full_path.path_join(scan_file);
			const String type = ResourceLoader::get_resource_type(path);
```

Reply:
Done and added `const` to a couple other Strings.